crates/iceberg/src/spec/partition.rs

```diff
@@ -131,50 +205,22 @@ impl PartitionSpec {
     /// * Field names
     /// * Source column ids
     /// * Transforms
-    pub fn is_compatible_with(&self, other: &UnboundPartitionSpec) -> bool {
+    pub fn is_compatible_with_unbound(&self, other: &UnboundPartitionSpec) -> bool {
```

Comment:
I moved the implementation to `BoundPartitionSpec` but comparing still to `SchemalessPartitionSpec` and not `BoundPartitionSpec`.

I need to use this method to check if a given spec already exists. The spec is passed as unbound but then bound against the `current_schema`. All old specs are however `SchemalessPartitionSpec ` and I would really like to avoid re-binding them all, as this is costly - we might have arbitrarily many schema specs as they are never cleaned u. Also the additional bind-check gives us no additional information. If the `BoundPartitionSpec` was able to bind against the schema, so will the `SchemalessPartitionSpec`.